apps/docs/content/guides/api/api-keys.mdx

```diff
@@ -1,66 +1,216 @@
+| `service_role`  | JWT (long lived)     | Elevated   | Platform, CLI | Exactly like secret keys.                                                                                                                                                                                                                                                                               |
 
-```sql
-create policy "Allow public access" on profiles to anon for
-select
-  using (true);
-```
+<Admonition type="note">
+
+`anon` and `service_role` keys are based on the project's JWT secret. They are generated when your project is created and can only be changed when you rotate the JWT secret. This can cause significant issues in production applications. Please use the publishable and secret keys instead.
+
+</Admonition>
+
+## `anon` and the publishable key
+
+Both the `anon` and publishable key authenticate public, insecure components of your application. These include:
+
+- Web pages, where the key is bundled in source code.
+- Mobile or desktop applications, where the key is bundled inside the compiled packages or executables.
+- CLI, scripts, tools, or other pre-built executables.
+- Other publicly available APIs that return the key without prior additional authorization.
+- Important: Turing test challenges, obfuscation, specialized knowledge does not count as authorization.
+
+These keys provide first-layer protection to your project's data, performance and bill, such as:
+
+- Providing basic Denial-of-Service protection, by requiring a minimal threshold of knowledge.
+- Protecting your bill by ignoring bots, scrapers, automated vulnerability scanners and other well meaning or random Internet activity.
+
+Since you often embed them in source code or build artifacts, it's easy to extract them with little or some specialized knowledge. You may wish to use obfuscation to increase this difficulty, but these approaches rarely produce the desired result.
+
+Importantly publishable and `anon` key **do not protect from**:
+
+- Static or dynamic code analysis and reverse engineering attempts.
+- Use of the Network inspector in the browser.
+- Cross-site request forgery, cross-site scripting, phishing attacks.
+- Man-in-the-middle attacks.
+- Others breaking the law on the Internet.
+
+When using the publishable or `anon` key, access to your project's data is guarded by Postgres via the built-in `anon` role. Unless you modify it, it will not allow reading or writing data to your project unless:
 
-And similarly for disallowing access:
+- You've modified the role's options to behave differently.
+- You've added a permissive Row-Level Security policy in your database granting acccess to the `anon` role.
```

Comment:
we could link to the hardening the data api docs here: https://supabase.com/docs/guides/database/hardening-data-api